src/BuildScriptGenerator/Python/PythonPlatform.cs

```diff
@@ -435,9 +435,29 @@ private string GetVirtualEnvironmentName(BuildScriptGeneratorContext context)
         private string GetMaxSatisfyingVersionAndVerify(string version)
         {
             var versionInfo = _versionProvider.GetVersionInfo();
+            var versionMap = versionInfo.SupportedVersions;
```

Comment:
Did you see your tests passing? i was suspecting this will not work since this property will _not_ have the preview versions in it since the parsing of the version `3.9.0b1` for example throws.

Reply:
Never mind. I remember now